FRs #1055/1073: Refactor Provider Identification, Host Provider, Invoice Issuer (#1139)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/metadata/data_generator/datagenerator.md

Reviewer: Starting to feel a little strange that this is `DataGenerator` and not `DataGeneratorName`........

Reviewer: We discussed this in a previous meeting.  Maybe we make the change in the future, but in the vein of minimizing surface area of the change, the author elected not to change this too, and that's good enough for me.  Resolving.

# Data Generator

> Human-readable name of the entity that generated the dataset instance, typically based on data provided by the [*Service Provider*](#serviceprovider). The Data Generator ensures the technical accuracy and delivery of the data.

Reviewer: ```suggestion
Human-readable name of the entity that generated the dataset instance, typically based on data provided by the [*Service Provider*](#serviceprovidername). The Data Generator ensures the technical accuracy and delivery of the data.
```

> Human-readable name of the entity responsible for producing the FOCUS dataset, typically based on data provided by the [*Service Provider*](#serviceprovider). The Data Generator ensures the technical accuracy and delivery of the data.

Reviewer 2: "Responsible" may be subjective. I think we're saying it's the person who *created* the dataset, right? My only issue is this doesn't clarify ammending the data. As a vendor, if I add a column, do I become the data generator? I would say so. I included "or last modified" even tho I don't love it. I'd love to find a better option.
```suggestion
Human-readable name of the entity that created or last modified the FOCUS dataset, typically based on data provided by the [*Service Provider*](#serviceprovider). The Data Generator ensures the technical accuracy and delivery of the data.
```

Author: I think if we get into who CRUD'ed (new word) the data we will get into more complexity. Ultimately it is whose is responsible. I would keep it at that.

Reviewer 2: But I'm not sure that's clear. Is the vendor responsible or the service provider who generated the data that the vendor updated? You could argue that a vendor is responsible for the data, regardless of whether the data from the service provider was bad or not. It's the vendor's decision to let the bad data through. From an end customer perspective, this gets into a game of pointing fingers, which is very subjective. Created or modified is objective.

Author: Take a look now. Moved it to reference the dataset instance and who generated it which is more accurate.

> Human-readable name of the entity responsible for producing the FOCUS dataset, typically based on data provided by the [*Service Provider*](#serviceprovider). The Data Generator ensures the technical accuracy and delivery of the data. 

Reviewer 3: ```suggestion
Human-readable name of the entity responsible for producing the FOCUS dataset, typically based on data provided by the [*Service Provider*](#serviceprovider). The Data Generator ensures the technical accuracy and delivery of the data.
```

Reviewer 3: @rileyjenk this is just a linter fix so I'll just go ahead and commit the change.

> The DataGenerator MUST be provided in the metadata. DataGenerator MUST be of type String and MUST NOT be null. The DataGenerator SHOULD be easily associated with the provider who generated the [*FOCUS dataset*](#glossary:FOCUS-dataset).

Reviewer: ```suggestion
* DataGenerator MUST be present in the metadata.
* DataGenerator MUST be of type String.
* DataGenerator MUST conform to [StringHandling](#stringhandling) requirements.
* DataGenerator MUST NOT be null.
* DataGenerator MUST be provided in the metadata. 
* DataGenerator SHOULD reflect the entity that generated the [*FOCUS dataset*](#glossary:FOCUS-dataset).
```
